tests/PactNet.Native.Tests/NativeRequestBuilderTests.cs

```diff
@@ -93,6 +93,14 @@ public void WithQuery_RepeatedQuery_SetsIndex()
             this.mockServer.Verify(s => s.WithQueryParameter(this.handle, "other", "value", 0));
         }
 
+        [Fact]
+        public void WithHeader_WhenCalledWithMatcher_AddsSerialisedHeaderParam()
```

Comment:
If we rename the existing test to `WithHeader_String_...` then this can be `WithHeader_Matcher_...`